src/Microsoft.NetCore.Analyzers/Core/Security/DoNotInstallRootCert.cs

```diff
@@ -0,0 +1,211 @@
+        {
+            switch (propertySetAbstractValue[0])
+            {
+                case PropertySetAbstractValueKind.Flagged:
+                    return HazardousUsageEvaluationResult.Flagged;
+
+                case PropertySetAbstractValueKind.Unflagged:
+                    return HazardousUsageEvaluationResult.Unflagged;
+
+                default:
+                    return HazardousUsageEvaluationResult.MaybeFlagged;
+            }
+        }
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+
+            // Security analyzer - analyze and report diagnostics on generated code.
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+
+            HazardousUsageEvaluatorCollection hazardousUsageEvaluators = new HazardousUsageEvaluatorCollection(
+                new HazardousUsageEvaluator("Add", HazardousUsageCallback));
+
+            context.RegisterCompilationStartAction(
+                (CompilationStartAnalysisContext compilationStartAnalysisContext) =>
+                {
+                    var wellKnownTypeProvider = WellKnownTypeProvider.GetOrCreate(compilationStartAnalysisContext.Compilation);
+
+                    if (!wellKnownTypeProvider.TryGetTypeByMetadataName(WellKnownTypeNames.SystemSecurityCryptographyX509CertificatesX509Store, out var x509TypeSymbol))
+                    {
+                        return;
+                    }
+
+                    if (!wellKnownTypeProvider.TryGetTypeByMetadataName(WellKnownTypeNames.SystemSecurityCryptographyX509CertificatesStoreName, out var storeNameTypeSymbol))
+                    {
+                        return;
+                    }
+
+                    // If X509Store is initialized with Root store, then that instance is flagged.
+                    var constructorMapper = new ConstructorMapper(
+                        (IMethodSymbol constructorMethod, IReadOnlyList<ValueContentAbstractValue> argumentValueContentAbstractValues,
+                        IReadOnlyList<PointsToAbstractValue> argumentPointsToAbstractValues) =>
+                        {
+                            var kind = PropertySetAbstractValueKind.Unflagged;
+
+                            if (constructorMethod.Parameters.Length > 0)
+                            {
+                                var valueContent = argumentValueContentAbstractValues[0].LiteralValues;
+
+                                if (constructorMethod.Parameters[0].Type.Equals(storeNameTypeSymbol) &&
+                                    valueContent.Contains(6) ||
+                                    constructorMethod.Parameters[0].Type.SpecialType == SpecialType.System_String &&
+                                    valueContent.Any(s => s.ToString().ToLower().Equals("root", StringComparison.Ordinal)))
```

Comment:
>s.ToString().ToLower().Equals("root", StringComparison.Ordinal) [](start = 58, length = 63)

How about String.Equals(s.ToString(), "root", StringComparison.OrdinalIgnoreCase)?  That'll avoid the extra string created from ToLower().